service/internal/builder/receivers_builder_test.go

```diff
@@ -334,7 +334,8 @@ func TestBuildReceivers_NotSupportedDataType(t *testing.T) {
 		t.Run(test.configFile, func(t *testing.T) {
 
 			cfg, err := configtest.LoadConfigFile(t, path.Join("testdata", test.configFile), factories)
-			assert.Error(t, err)
```

Comment:
This was changed in https://github.com/open-telemetry/opentelemetry-collector/pull/2802 because NopConfig.Validate was failing because "bf" (badfactory) was not the same type as "nop"

Reply:
It was my bad. Thx for correcting it.